packages/rum/src/domain/record/privacy.spec.ts

```diff
@@ -17,147 +12,152 @@ describe('privacy helpers', () => {
-      expect(getNodeInputPrivacyMode(node, InputPrivacyMode.MASKED)).toBe(InputPrivacyMode.MASKED)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.ALLOW)).toBe(NodePrivacyLevel.ALLOW)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.IGNORE)).toBe(NodePrivacyLevel.IGNORE)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.MASK)).toBe(NodePrivacyLevel.MASK)
     })
 
     it('use the ancestor privacy mode for a DOM HTMLInputElement with a type of "text"', () => {
       const node = document.createElement('input')
       node.type = 'text'
-      expect(getNodeInputPrivacyMode(node, InputPrivacyMode.NONE)).toBe(InputPrivacyMode.NONE)
-      expect(getNodeInputPrivacyMode(node, InputPrivacyMode.IGNORED)).toBe(InputPrivacyMode.IGNORED)
-      expect(getNodeInputPrivacyMode(node, InputPrivacyMode.MASKED)).toBe(InputPrivacyMode.MASKED)
-    })
-
-    it('considers a DOM Document as not to be ignored', () => {
-      expect(getNodeInputPrivacyMode(document, InputPrivacyMode.NONE)).toBe(InputPrivacyMode.NONE)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.ALLOW)).toBe(NodePrivacyLevel.ALLOW)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.IGNORE)).toBe(NodePrivacyLevel.IGNORE)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.MASK)).toBe(NodePrivacyLevel.MASK)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.MASK_FORMS_ONLY)).toBe(NodePrivacyLevel.MASK)
     })
 
     it('considers a DOM Element with a data-dd-privacy="input-ignored" attribute to be ignored', () => {
       const node = document.createElement('input')
       node.setAttribute('data-dd-privacy', 'input-ignored')
-      expect(getNodeInputPrivacyMode(node, InputPrivacyMode.NONE)).toBe(InputPrivacyMode.IGNORED)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.NOT_SET)).toBe(NodePrivacyLevel.MASK)
     })
 
-    it('considers a DOM Element with a dd-privacy-input-ignored class to be ignored', () => {
+    it('considers a DOM Element with a dd-privacy-input-ignored class to be input-ignored (mask-froms-only)', () => {
       const node = document.createElement('input')
       node.className = 'dd-privacy-input-ignored'
-      expect(getNodeInputPrivacyMode(node, InputPrivacyMode.NONE)).toBe(InputPrivacyMode.IGNORED)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.NOT_SET)).toBe(NodePrivacyLevel.MASK)
     })
 
-    it('considers a DOM HTMLInputElement with a type of "password" to be ignored', () => {
+    it('considers a DOM HTMLInputElement with a type of "password" to be MASK', () => {
       const node = document.createElement('input')
       node.type = 'password'
-      expect(getNodeInputPrivacyMode(node, InputPrivacyMode.NONE)).toBe(InputPrivacyMode.IGNORED)
+      expect(getNodePrivacyLevel(node, NodePrivacyLevelInternal.NOT_SET)).toBe(NodePrivacyLevel.MASK)
     })
 
     describe('input mode priority', () => {
```

Comment:
Although the product spec has changed (priority + aliased), documenting the deprecation approach with tests seems like a good idea, no? In the worst case, we've got a minimally redundant test.

Reply:
Mh I see, it does make sense since we have about the same priority rules with IGNORE/HIDDEN. But I would still adjust them a bit:


* the first test does not make much sense since input-ignored and input-masked are now the same
* the second test is fine, but should be tested with non-deprecated values too (collocate test cases)
* the third test could be reworded with something like: `consider a DOM element to be ALLOW even if an ancestor is MASK`

Reply:
1. now uses mask-forms-only
2. added mask-forms-only also
3. changed description